extensions/amp-carousel/0.2/amp-carousel.js

```diff
@@ -632,6 +636,19 @@ class AmpCarousel extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Attach singal onto touchmove event to stop propagation to viewer
+   * to be handled at the viewer integration layer in bubble phase.
+   */
+  handlePropagationInViewer_() {
+    if (!this.isViewerEmbeded_) {
```

Comment:
This is much cleaner than the version we discussed earlier (using capture phase of the event to pass this information). Nice job!